great_expectations_cloud/agent/actions/generate_data_quality_check_expectations_action.py

```diff
@@ -94,39 +96,50 @@ def run(self, event: GenerateDataQualityCheckExpectationsEvent, id: str) -> Acti
                 created_resources.append(
                     CreatedResource(resource_id=str(metric_run_id), type="MetricRun")
                 )
-                if DataQualityIssues.VOLUME in selected_dqi:
-                    constraint_fn = "forecast" if event.use_forecast else "mean"
-                    volume_change_expectation_id = self._add_volume_change_expectation(
-                        asset_id=data_asset.id,
-                        constraint_fn=constraint_fn,
-                    )
-                    created_resources.append(
-                        CreatedResource(
-                            resource_id=str(volume_change_expectation_id), type="Expectation"
-                        )
-                    )
 
-                if DataQualityIssues.SCHEMA in selected_dqi:
-                    schema_change_expectation_id = self._add_schema_change_expectation(
-                        metric_run=metric_run, asset_id=data_asset.id
+                if selected_dqi:
+                    pre_existing_anomaly_detection_coverage = (
+                        self._get_current_anomaly_detection_coverage(data_asset.id)
                     )
-                    created_resources.append(
-                        CreatedResource(
-                            resource_id=str(schema_change_expectation_id), type="Expectation"
+
+                    if (
+                        DataQualityIssues.VOLUME in selected_dqi
+                        and DataQualityIssues.VOLUME not in pre_existing_anomaly_detection_coverage
+                    ):
+                        constraint_fn = "forecast" if event.use_forecast else "mean"
```

Comment:
Can we define an enum for the constraint function types with `forecast` and `mean` and then use that throughout this file.